lib/Analysis/AxisInfo.cpp

```diff
@@ -1074,8 +1082,8 @@ LogicalResult AxisInfoAnalysis::visitOperation(
     auto vals = cast<DenseElementsAttr>(attr).getValues<int>();
     newConstancy = AxisInfo::DimVectorT(vals.begin(), vals.end());
   }
-  curr = AxisInfo(newContiguity, newDivisibility, newConstancy,
-                  curr.getConstantValue());
+  curr = AxisInfo(std::move(newContiguity), std::move(newDivisibility),
+                  std::move(newConstancy), curr.getConstantValue());
```

Comment:
I'm not sure those move semantic do much as the constructor currently takes arg by copy. Is there anything special you are trying to fix? I'm not sure this helps readability

Reply:
> constructor currently takes arg by copy

The `AxisInfo` constructor takes arguments by their value, yes, in this case `SmallVector`' copy constructor or the move constructor  can be called (`SmallVector` has a move constructor). Since `std::move` casts the object to something like: `SmallVector&&`, the move constructor will be called, that is, we can get rid of one copy (for each `SmallVector` value).

> I'm not sure this helps readability

It's a bit harder to read, I agree, but it saves three copies.

Reply:
Isn't the better fix here simply making the AxisInfo constructor take everything by const-ref? I think it's just an oversight, as the constructor doesn't use the "copy-move" idiom.

Reply:
> Isn't the better fix here simply making the AxisInfo constructor take everything by const-ref? I think it's just an oversight, as the constructor doesn't use the "copy-move" idiom.

I agree

Reply:
Or `ArrayRef<int64_t>`

Reply:
Done